app/ui.js

```diff
@@ -1099,14 +1096,49 @@ var UI;
             // Don't display the connection settings until we're actually disconnected
         },
 
+        reconnect: function() {
+            delete UI.reconnect_callback;
+
+            // if reconnect has been disabled in the meantime, do nothing.
+            if (UI.inhibit_reconnect) {
+                return;
+            }
+
+            UI.connect();
+        },
+
         disconnectFinished: function (rfb, reason) {
+            var reconnect = UI.getSetting('reconnect');
+
+            if ((reconnect === 'true' || reconnect == '1') && !UI.inhibit_reconnect) {
```

Comment:
The rest of the code just checks booleans directly without doing string comparisons, so it's probably best to do the same thing here.